docs/spec/api/0.1/recommendations_for_activity_streams.md

```diff
@@ -370,7 +370,7 @@ The _Entry Point_{:.term} _MAY_{:.strong-term} have a _totalItems_{:.term} prope
 Reference: [Ordered Collection Page][org-w3c-activitystreams-coretype-orderedcollectionpage] in the [Activity Stream specification][org-w3c-activitystreams]
 {:.reference}
 
-Each time a set of changes is published, changes _MUST_{:.strong-term} be released in at least one _Change Set_{:.term}. Changes _MAY_{:.strong-term} be published across multiple _Change Sets_{:.term}. For example, a site may decide that each _Change Set_{:.term} will have at most 50 changes and if that maximum is exceeded during the release time period, then a second _Change Set_{:.term} will be created. All changes within a _Change Set_{:.term} and, if applicable, across  Change Sets _MUST_{:.strong-term} be sorted in date-time order in the _orderedItems_{:.term} property with the earliest change in the set appearing first and most recent change in the set appearing last.
+Each time a set of changes is published, changes _MUST_{:.strong-term} be released in at least one _Change Set_{:.term}.  Changes _MAY_{:.strong-term} be published across multiple _Change Sets_{:.term}.  For example, a site may decide that each _Change Set_{:.term} will have at most 50 changes and if that maximum is exceeded during the release time period, then a second _Change Set_{:.term} will be created. It is recommended that all changes within a _Change Set_{:.term} and, if applicable, across Change Sets be sorted in date-time order in the _orderedItems_{:.term}. For practical reasons, _Change Set_{:.term} and corresponding _Activities_{:.term} _MAY_{:.strong-term} be in ascending or descending order, but the order _MUST_{:.strong-term} be consistent within the _Entity Metadata Collection_{:.term}. Acknowledging not all implementations will store every change for an entity over time, _Entity Metadata Collection_{:.term} _MAY_{:.strong-term} provide feeds of only the last known metadata update for an entity. In these cases, because the _Entity Metadata Collection_{:.term} is not persistent (meaning _Ordered Collection Pages_{:.term} are overwritten, providing only the latest updates), _Activities_{:.term} within the _Entity Metadata Collection_{:.term} must include a date using _property_{:.term} because the page number can not be used to know the last _Activities_{:.term} processed by a consumer.
```

Comment:
I think the new formulation ends up being confusing and not saying what we intend. I think we need to give separate instructions for Change Sets and for changes within those sets. Perhaps:

> Each time a set of changes is published, the changes MUST be released as _Activities_ in at least one _Change Set_.  Changes _MAY_ be published across multiple _Change Sets_.  For example, a site may decide that each _Change Set_ will have at most 50 changes and if that maximum is exceeded during the release time period, then a second _Change Set_ will be created.
> 
> * The _Activities_ within a _Change Set_ SHOULD be sorted in date-time order in the `orderedItems` array. The _Activities_  MAY be in ascending or descending order, but the order _MUST_ be consistent within the _Entity Metadata Collection_.
> * Where there are multiple _Change Sets_, these sets MUST be arranged in ascending date-time order. That is, whatever order _Activities_ appear in inside the _Change Sets_, all of activities in the first _Change Set_ must be  before or at the same time as those in the second _Change Set_, and so on.
>
> Not all implementations will store every change for an entity over time. An _Entity Metadata Collection_ _MAY_ provide feeds of only the last known metadata update for an entity. In this case, because the _Entity Metadata Collection_ is not persistent (meaning _Ordered Collection Pages_ are overwritten, providing only the latest updates), _Activities_ within the _Entity Metadata Collection_ MUST include a date using the `updated` property because the page identifier cannot be used to know the last _Activities_ processed by a consumer.

I also note that we should not use "recommended" unless done in RFC2119 style, and I would recommend (he he) that try to avoid it anyway, better to decide between truly optional (OPTIONAL or MAY) and SHOULD.

Reply:
Updates, based on 2022-12-19...

Each time a set of changes is published, the changes MUST be released as Activities in at least one Change Set. Changes MAY be published across multiple Change Sets. For example, a site may decide that each Change Set will have at most 50 changes and if that maximum is exceeded during the release time period, then a second Change Set will be created.

- The Activities within a Change Set MUST be sorted in date-time order in the orderedItems array. The Activities MAY be in ascending or descending order, but the order MUST be consistent within the Entity Metadata Collection.

- Where there are multiple Change Sets, these sets MUST be arranged in ascending or descending date-time order, consistent with the Activity ordering within a given Change Set.

Not all implementations will store every change for an entity over time. An Entity Metadata Collection MAY provide feeds of only the last known metadata update for an entity. In this case, because the Entity Metadata Collection is not persistent (meaning Ordered Collection Pages are overwritten, providing only the latest updates), Activities within the Entity Metadata Collection MUST include a date using the updated property because the page identifier cannot be used to know the last Activities processed by a consumer.

Reply:
Replace "Entity Metadata Collection" with "Collection"

Reply:
Have incorporated the changes discussed in [7beafd4](https://github.com/LD4/entity_metadata_management/pull/73/commits/7beafd4ea8838742d5ae3e254d54ea0d7032f15f)